avr-hal-generic/src/port.rs

```diff
@@ -781,13 +781,18 @@ macro_rules! renamed_pins {
             type McuPins = $mcu_pins:ty;
         }
     ) => {
-        $(#[$pins_attr])*
-        pub struct Pins {
-            $($(#[$pin_attr])*
-            pub $pin_name: $pin_wrapper<
-                $crate::port::mode::Input<$crate::port::mode::Floating>,
-                $pin_type,
-            >,)+
+        $crate::paste::paste! {
+            $(#[$pins_attr])*
+            pub struct Pins {
+                    $(pub $pin_name: [<Pin $pin_name:upper>]<
+                        $crate::port::mode::Input<$crate::port::mode::Floating>,
+                    >,)+
+            }
+        }
+
+        $crate::paste::paste! {
+            $($(#[$pin_attr])* pub type [<$pin_name:upper>] = $pin_type;)+
+            $(#[doc = " Safely wrapped version of:"] $(#[$pin_attr])* pub type [<Pin $pin_name:upper>]<MODE> = $pin_wrapper<MODE, $pin_type>;)+
```

Comment:
Now that you mentioned it, I think this second type should be dropped.  Only keeping what you called the "raw type" seems more consistent to me.